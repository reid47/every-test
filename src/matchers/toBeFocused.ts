export function toBeFocused(received) {
  const pass = document.activeElement === received;

  const message = pass
    ? 'Expected element not to be focused, but it was.'
    : 'Expected element to be focused, but it was not.';

  return { pass, message };
}
