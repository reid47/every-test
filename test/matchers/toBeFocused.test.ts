import toBeFocused from '../../src/matchers/toBeFocused';

describe('toBeFocused', () => {
  test('passing', () => {
    const el = document.createElement('input');
    el.focus();

    const result = toBeFocused(el);
    expect(result.pass).toBeTruthy();
    expect(result.message).toMatchSnapshot();
  });

  test('failing', () => {
    const el = document.createElement('input');

    const result = toBeFocused(el);
    expect(result.pass).toBeFalsy();
    expect(result.message).toMatchSnapshot();
  });
});
