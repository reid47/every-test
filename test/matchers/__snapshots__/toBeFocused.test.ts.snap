// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeFocused failing 1`] = `"Expected element to be focused, but it was not."`;

exports[`toBeFocused passing 1`] = `"Expected element not to be focused, but it was."`;
